src/encoded/static/components/encyclopedia.js

```diff
@@ -1,49 +1,492 @@
+        query.deleteKeyValue('biosample_ontology');
+        // All file options are selected
+        if (newFiles.length === fileOptions.length) {
+            query.addKeyValue('annotation_type', 'representative DNase hypersensitivity sites');
+            query.addKeyValue('annotation_type', 'candidate Cis-Regulatory Elements');
+            query.addKeyValue('biosample_ontology', '*', true);
+        // Only rDHS is selected
+        } else if (newFiles.length === 1 && newFiles[0] === 'rDHS') {
+            query.addKeyValue('annotation_type', 'representative DNase hypersensitivity sites');
+        // rDHS and other annotation subtypes are selected
+        // This is the option that does not currently work *****
+        } else if (newFiles.indexOf('rDHS') > -1) {
+            query.addKeyValue('annotation_type', 'representative DNase hypersensitivity sites');
+            query.addKeyValue('annotation_type', 'candidate Cis-Regulatory Elements');
+            query.addKeyValue('biosample_ontology', '*', true);
+            newFiles.forEach((type) => {
+                if (type !== 'rDHS') {
+                    query.addKeyValue('annotation_subtype', type);
+                }
+            });
+        // Only non-rDHS annotation subtypes are selected
+        } else {
+            newFiles.forEach((type) => {
+                query.addKeyValue('annotation_subtype', type);
+            });
+        }
+        const newHref = query.format();
+        setSelectedFiles(newFiles);
+        setdownloadHref(newHref);
+    };
+
+    // Update assembly, organism, browser files, and download link when user clicks on tab
+    const handleTabClick = (tab) => {
+        setVizFiles([]);
+        setSelectedOrganism(tab);
+        setAssembly(defaultAssemblyByOrganism[tab]);
+
+        // Update download link
+        const query = new QueryString(downloadHref);
+        query.replaceKeyValue('assembly', defaultAssemblyByOrganism[tab]);
+        const newHref = query.format();
+        setdownloadHref(newHref);
+    };
+
+    // Annotation type facet is not a normal facet, at least one annotation must be selected
+    // Facet filters look like radiobuttons
+    const chooseAnnotationType = (type) => {
+        let newAnnotationType = annotationType;
+        if (annotationType.indexOf(type) > -1 && annotationType.length > 1) {
+            newAnnotationType = annotationType.filter((annotation) => annotation !== type);
+        } else if (annotationType.indexOf(type) === -1) {
+            newAnnotationType = [...annotationType, type];
+        }
+        setAnnotationType(newAnnotationType);
+    };
+
+    // Facet filters are redirected to update the urls that populate the browser
+    const handleFacetClick = (e) => {
+        // If clicked filter belongs to the annotation facet, clear filters, or boolean switch, we do not want to prevent default
+        const isButton = findUpTag(e.target, 'BUTTON');
```

Comment:
Element-sniffing (checking for element type) seems like a code smells and makes the code brittle. Like how about if someone uses -

`<input type"button">`

or adds some other element you did not cover..? I may be wrong on this.

Reply:
Well, this is all not that ideal - redirecting links is not the greatest - but we're bypassing the API so there's not much else we can do. If someone adds a new elements we haven't accounted for, this may need an update, but as it is, the buttons are from the special facets that I've added and the links are from the default facets, so it should be ok.